### ISSUE TITLE
parallel perfomance

### ISSUE THREAD

**Issue by AshKelly** [syasini/AstroPaint#81]
I find that in my local example problems (which are mostly derivatives of those given in the docs) that `n_cpus=1` is **faster** than any other choice.

However, the code defaults to `n_cpus=-1` for all problems. Do you expect / also find that serial tasks should be faster? Do you expect a particular problem size at which parallel jobs are faster?

If this result is typical, perhaps `n_cpus=1` is a better default, or a value which can switch depending on the inferred problem size. 

This is just little thought I had - and maybe, my experience is isolated and the whole issue is redundant.

**Comment by syasini** [comment]
Great catch! You're absolutely right. For smaller size catalogs the overhead from parallelization makes the code run slower so it makes sense to use n_cpus=1 as default. I'll modify this on `develop` and post an update here as soon as I'm done.

**Comment by syasini** [comment]
I changed the default value of `n_cpus` to 1 and also changed the corresponding lines in the README file to reflect this. 

https://github.com/syasini/AstroPaint/blob/develop/astropaint/paint_bucket.py#L2299

If you think this is good I'll merge it to `master`.

**Comment by AshKelly** [comment]
I certainly think `n_cpus=1` default is better. For two reasons: 1) for small problems when learning the library, I found that parallel mode was up to 2x slower. 2) when using this on an HPC login node, using all the CPUs is frowned upon, so it's a little worrying that I could do this by accident.

BTW, I did also run a few larger problems where the parallel mode was significantly faster, so I'm certainly glad it exists!
